src/Operations.jl

```diff
@@ -1661,10 +1661,27 @@ function test(ctx::Context, pkgs::Vector{PackageSpec};
             printpkgstyle(ctx.io, :Testing, "Running tests...")
             flush(ctx.io)
             cmd = gen_test_code(testfile(source_path); coverage=coverage, julia_args=julia_args, test_args=test_args)
-            p = run(pipeline(ignorestatus(cmd), stdout = sandbox_ctx.io))
+            p = run(pipeline(ignorestatus(cmd), stdin = stdin, stdout = sandbox_ctx.io, stderr = stderr_f()), wait = false)
+            interrupted = false
+            try
+                wait(p)
+            catch e
+                if e isa InterruptException
+                    interrupted = true
+                    print("\n")
+                    printpkgstyle(ctx.io, :Testing, "Tests interrupted. Exiting the test process\n", color = Base.error_color())
+                    # Give some time for the child interrupt handler to print a stacktrace and exit,
+                    # then kill the process if still running
+                    if timedwait(() -> !process_running(p), 4) == :timed_out
```

Comment:
Using `timedwait` is often a general design problem, where you can usually use a `Timer(4)` and `wait(p)` instead though